paddle/utils/Error.h

```diff
@@ -116,6 +116,8 @@ class Error {
    */
   operator bool() const { return msg_ == nullptr; }
 
+  bool isOK() const { return *this; }
+
```

Comment:
Here is a problem.

I think the origin `operator bool() const` we defined in `paddle::Error` is very confusing in actually use.

The sample usage are:

```cpp
Error err = somefunc();

if (!err) {  // Error occurred.
   ...
}
```

it seems `if(!err)` should mean `if there is no error`, not `if there is not ok`. 

So I temporarily add `isOK()` method, to clarify the meaning. And should we changed the original `operator bool()const` method? @wangkuiyi